src/ai/backend/manager/models/session.py

```diff
@@ -208,43 +208,37 @@ class SessionStatus(enum.Enum):
 
 SESSION_STATUS_TRANSITION_MAP: Mapping[SessionStatus, set[SessionStatus]] = {
     SessionStatus.PENDING: {
-        s for s in SessionStatus if s not in (SessionStatus.PENDING, SessionStatus.TERMINATED)
+        SessionStatus.SCHEDULED,
+        SessionStatus.TERMINATING,
+        SessionStatus.TERMINATED,
+        SessionStatus.ERROR,
+        SessionStatus.CANCELLED,
     },
     SessionStatus.SCHEDULED: {
-        s
-        for s in SessionStatus
-        if s
-        not in (
-            SessionStatus.SCHEDULED,
-            SessionStatus.PENDING,
-            SessionStatus.TERMINATED,
-            SessionStatus.CANCELLED,
-        )
+        SessionStatus.PULLING,
+        SessionStatus.PREPARING,
+        SessionStatus.TERMINATED,
+        SessionStatus.ERROR,
+        SessionStatus.CANCELLED,
     },
     SessionStatus.PULLING: {
-        s
-        for s in SessionStatus
-        if s
-        not in (
-            SessionStatus.PULLING,
-            SessionStatus.PENDING,
-            SessionStatus.SCHEDULED,
-            SessionStatus.TERMINATING,  # cannot destroy PULLING session
-            SessionStatus.TERMINATED,
-            SessionStatus.CANCELLED,
-        )
+        SessionStatus.PREPARING,
+        SessionStatus.RUNNING,
+        SessionStatus.RUNNING_DEGRADED,
+        # SessionStatus.TERMINATING,  # cannot destroy PULLING session by user
+        SessionStatus.TERMINATED,
+        SessionStatus.ERROR,
+        SessionStatus.CANCELLED,
     },
     SessionStatus.PREPARING: {
-        s
-        for s in SessionStatus
-        if s
-        not in (
-            SessionStatus.PREPARING,
-            SessionStatus.PENDING,
-            SessionStatus.SCHEDULED,
-            SessionStatus.TERMINATED,
-            SessionStatus.CANCELLED,
-        )
+        SessionStatus.PULLING,
+        SessionStatus.RUNNING,
+        SessionStatus.RESTARTING,
```

Comment:
RESSTARTING should be removed from here because RESTARTING may occur only with RUNNING sessions.

Reply:
Okay, I removed this part!